python/cudf_polars/tests/expressions/test_datetime_basic.py

```diff
@@ -37,24 +40,105 @@ def test_datetime_dataframe_scan(dtype):
     assert_gpu_result_equal(query)
 
 
+datetime_extract_fields = [
+    "year",
+    "month",
+    "day",
+    "weekday",
+    "hour",
+    "minute",
+    "second",
+    "millisecond",
+    "microsecond",
+    "nanosecond",
+]
+
+unsupported_extract_fields = ["millennium"]
+
+
+@pytest.fixture(
+    ids=datetime_extract_fields,
+    params=[methodcaller(f) for f in datetime_extract_fields],
+)
+def field(request):
+    return request.param
+
+
+@pytest.fixture(
+    ids=unsupported_extract_fields,
+    params=[methodcaller(f) for f in unsupported_extract_fields],
+)
+def unsupported_field(request):
+    return request.param
+
+
+def test_datetime_extract(field):
+    ldf = pl.LazyFrame(
+        {
+            "datetimes": pl.datetime_range(
+                datetime.datetime(2020, 1, 1),
+                datetime.datetime(2021, 12, 30),
+                "3mo14h15s11ms33us999ns",
+                eager=True,
+            )
+        }
+    )
+
+    q = ldf.select(field(pl.col("datetimes").dt))
+
+    with pytest.raises(AssertionError):
+        # polars produces int32, libcudf produces int16 for extraction methods
```

Comment:
Can’t we just upcast the libcudf results? This seems in scope for this PR.

Reply:
We closed the below-linked cudf issue since `std::chrono` specifies int16 as the width of the year extraction function.

Also, this comment is wrong, because for the new ones you've implemented, you're returning an int64 when you absolutely _could_ be returning an int32.

Reply:
This should be addressed on the IR side now.